nano/secure/ledger_set_any.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <nano/secure/account_iterator.hpp>
+#include <nano/secure/receivable_iterator.hpp>
+
+#include <optional>
+
+namespace nano
+{
+class account_info;
+class block;
+class block_hash;
+class ledger;
+class pending_info;
+class pending_key;
+class qualified_root;
+}
+namespace nano::store
+{
+class transaction;
+}
+
+namespace nano
+{
+class ledger_set_any
+{
+public:
+	using account_iterator = nano::account_iterator<ledger_set_any>;
+	using receivable_iterator = nano::receivable_iterator<ledger_set_any>;
+
+	ledger_set_any (nano::ledger const & ledger);
+
+	std::optional<nano::account> account (store::transaction const & transaction, nano::block_hash const & hash) const;
+	account_iterator account_begin (store::transaction const & transaction) const;
+	account_iterator account_end () const;
+	// Returns the next account entry equal or greater than 'account'
+	account_iterator account_lower_bound (store::transaction const & transaction, nano::account const & account) const;
+	// Returns the next account entry greater than 'account'
+	account_iterator account_upper_bound (store::transaction const & transaction, nano::account const & account) const;
+	std::optional<nano::uint128_t> amount (store::transaction const & transaction, nano::block_hash const & hash) const;
```

Comment:
It is not clear what guarantees this function gives.
For example, in what scenario will a previous block not have a balance and hence an amount will not be returned?
<img width="743" alt="image" src="https://github.com/nanocurrency/nano-node/assets/3601511/16ce2535-18a5-4248-ade4-b62079369619">

Reply:
I agree these are confusing, they're to deal with block pruning.  I tried to simplify this but the only way I could fix the issue would be to store the amount as sideband which might be a good idea but requires a block table upgrade.

This matches existing behavior.